ivy/functional/backends/paddle/experimental/manipulation.py

```diff
@@ -162,6 +162,12 @@ def hstack(
             return ivy.concat(arrays, axis=0)
 
 
+with_supported_device_and_dtypes(
+    {"2.5.1 and below": {"cpu": ("bool", "int32", "int64", "float32", "float64")}},
```

Comment:
> x (Tensor) – The input Tensor(or LoDTensor). The data type of the input Tensor x should be float16, float32, float64, int32, int64, bool. float16 is only supported on gpu.

Dtypes `bool` and `float16` are also supported on CPU and GPU respectively, please make the required changes in the decorator.

Reply:
I have added `bool` to CPU already and I will add `float16` to GPU. Thank you for your response

Reply:
@sagewhocodes I have pushed the asked changes kindly review.